From b97676124fbe4909d78249cbc9cec416275331c7 Mon Sep 17 00:00:00 2001
From: "Niels Westphal (crashniels)" <nielswest@live.de>
Date: Sat, 21 Jun 2025 16:54:46 +0200
Subject: [PATCH] Use msi_alloc_desc instead
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="------------false"

This is a multi-part message in MIME format.
--------------false
Content-Type: text/plain; charset=UTF-8; format=fixed
Content-Transfer-Encoding: 8bit


msi_alloc_desc is the rename of the older function we used to use called alloc_msi_entry.
---
 drivers/ps4/ps4-apcie.c | 9 ++-------
 include/linux/msi.h     | 3 +++
 kernel/irq/msi.c        | 2 +-
 3 files changed, 6 insertions(+), 8 deletions(-)


--------------false
Content-Type: text/x-patch; name="0017-Use-msi_alloc_desc-instead.patch"
Content-Transfer-Encoding: 8bit
Content-Disposition: attachment; filename="0017-Use-msi_alloc_desc-instead.patch"

diff --git a/drivers/ps4/ps4-apcie.c b/drivers/ps4/ps4-apcie.c
index fc5582518c80b484b66cd2aac9e09f029502b47e..cd332c4db65553ac8576e48c94e2133227cbcd7a 100644
--- a/drivers/ps4/ps4-apcie.c
+++ b/drivers/ps4/ps4-apcie.c
@@ -330,13 +330,6 @@ int apcie_assign_irqs(struct pci_dev *dev, int nvec)
 	struct msi_desc *desc;
 	struct device* bare_dev = &sc->pdev->dev;
 
-	desc = kzalloc(sizeof(*desc), GFP_KERNEL);
-	if (!desc)
-    	return -ENOMEM;
-
-	desc->nvec_used = nvec;
-	desc->dev = bare_dev;
-
 	/* Our hwirq number is function << 8 plus subfunction.
 	 * Subfunction is usually 0 and implicitly increments per hwirq,
 	 * but can also be 0xff to indicate that this is a shared IRQ. */
@@ -349,6 +342,8 @@ int apcie_assign_irqs(struct pci_dev *dev, int nvec)
 	}
 #endif
 
+	desc = msi_alloc_desc(bare_dev, nvec, NULL);
+
 	info.desc = desc;
 	info.data = sc;
 
diff --git a/include/linux/msi.h b/include/linux/msi.h
index fa41eed6286826dbc6583002001561ab816f08c2..79d1bd76f9b53f23b46aedcc3fb706c1701473aa 100644
--- a/include/linux/msi.h
+++ b/include/linux/msi.h
@@ -239,6 +239,9 @@ DEFINE_LOCK_GUARD_1(msi_descs_lock, struct device, __msi_lock_descs(_T->lock),
 struct msi_desc *msi_domain_first_desc(struct device *dev, unsigned int domid,
 				       enum msi_desc_filter filter);
 
+struct msi_desc *msi_alloc_desc(struct device *dev, int nvec,
+				       const struct irq_affinity_desc *affinity);
+
 /**
  * msi_first_desc - Get the first MSI descriptor of the default irqdomain
  * @dev:	Device to operate on
diff --git a/kernel/irq/msi.c b/kernel/irq/msi.c
index 3cafa40e6ce3429df2434903dc383df556311301..0cb70a500e846697917f5eb6f7d673dbaa427d2f 100644
--- a/kernel/irq/msi.c
+++ b/kernel/irq/msi.c
@@ -73,7 +73,7 @@ static int msi_domain_prepare_irqs(struct irq_domain *domain, struct device *dev
  *
  * Return: pointer to allocated &msi_desc on success or %NULL on failure
  */
-static struct msi_desc *msi_alloc_desc(struct device *dev, int nvec,
+struct msi_desc *msi_alloc_desc(struct device *dev, int nvec,
 				       const struct irq_affinity_desc *affinity)
 {
 	struct msi_desc *desc = kzalloc_obj(*desc);

--------------false--


